File: Algorithms/Sorting/selection_sort.py
def selection_sort(arr):

    for i in range(len(arr)):
        min = i # index of min elem

        for j in range(i+1,len(arr)):
            # arr[j] is smaller than the min elem
            if arr[j] < arr[min]:
                min = j
             
        arr[i],arr[min] = arr[min],arr[i] # swap   
        # print(arr)
    
    return arr

File: Algorithms/Sorting/SelectionSort.java
public class SelectionSort {  
    public static void selectionSort(int[] arr){  
        for (int i = 0; i < arr.length - 1; i++)  
        {  
            int index = i;  
            for (int j = i + 1; j < arr.length; j++){  
                if (arr[j] < arr[index]){  
                    index = j;//searching for lowest index  
                }  
            }  
            int smallerNumber = arr[index];   
            arr[index] = arr[i];  
            arr[i] = smallerNumber;  
        }  
    }  
       
    public static void main(String a[]){  
        int[] arr1 = {49, 14, 15, 82, 1, 206, 352, 56};  
        System.out.println("Before Selection Sort");  
        for(int i:arr1){  
            System.out.print(i+" ");  
        }  
        System.out.println();  
          
        selectionSort(arr1);//sorting array using selection sort  
         
        System.out.println("After Selection Sort");  
        for(int i:arr1){  
            System.out.print(i+" ");  
        }  
    }  
}
